packages/manager/modules/pci/src/projects/project/storages/databases/components/dashboard-integrations/dashboard-integrations.controller.js

```diff
@@ -0,0 +1,20 @@
+export default class {
+  $onInit() {
+    this.integrationTypes = this.integrations
+      .reduce((integrationsByType, integration) => {
+        const integrationType = integrationsByType.find(
+          ({ type }) => type === integration.type,
+        );
+        if (integrationType) {
+          integrationType.count += 1;
+        } else {
+          integrationsByType.push({
+            type: integration.type,
+            count: 1,
+          });
+        }
+        return integrationsByType;
+      }, [])
```

Comment:
**suggestion(perf):** We can avoid looping inside a loop (`Array.find inside Array.reduce`) by leveraging `objects`.
```suggestion
    this.integrationTypes = this.integrations
      .reduce((integrationsByType, integration) => {
        if (integrationsByType[integration.type]) {
         integrationsByType[integration.type] += 1;
        } else {
         integrationsByType[integration.type] = 1;
        }
        return integrationsByType;
      }, {})
```
Then convert to array if required.

Reply:
It does work, however it won't pass linter because of the no-param-reassign rule.